apps/condo/domains/contact/tasks/exportContacts.js

```diff
@@ -0,0 +1,166 @@
+    }
+}
+
+const buildExportFile = async ({ rows, task }) => {
+    const { where, timeZone, locale, id } = task
+    const createdAtGte = get(findAllByKey(where, 'createdAt_gte'), 0)
+    const createdAtLte = get(findAllByKey(where, 'createdAt_lte'), 0)
+
+    const formatDate = (date) => dayjs(date).tz(timeZone).format(DATE_FORMAT)
+
+    let headerMessage = i18n('excelExport.header.contacts.forAllTime', { locale })
+    if (createdAtGte && createdAtLte) {
+        headerMessage = `${i18n('excelExport.header.contacts.forPeriod', { locale })} ${formatDate(createdAtGte)} — ${formatDate(createdAtLte)}`
+    }
+
+    const { stream } = await buildExportExcelFile({
+        templatePath: './domains/contact/templates/ContactsExportTemplate.xlsx',
+        replaces: {
+            header: headerMessage,
+            contacts: rows,
+            i18n: {
+                ...getHeadersTranslations(EXPORT_TYPE_CONTACTS, locale),
+                sheetName: i18n('excelExport.sheetNames.contacts', { locale }),
+            },
+        },
+    })
+
+    return {
+        stream,
+        filename: `contacts_${dayjs().format('DD_MM')}.xlsx`,
+        mimetype: EXCEL_FILE_META.mimetype,
+        encoding: EXCEL_FILE_META.encoding,
+        meta: {
+            listkey: 'ContactExportTask',
+            id,
+        },
+    }
+}
+
+async function exportContacts (taskId) {
+    if (!taskId) throw new Error('taskId is required')
+    const { keystone: context } = await getSchemaCtx('ContactExportTask')
+
+    const task = await ContactExportTask.getOne(context, { id: taskId })
+    const { format, where, sortBy } = task
+
+    const baseAttrs = { dv: 1, sender: TASK_DV_AND_SENDER }
+
+    if (!task.locale) {
+        await ContactExportTask.update(context, taskId, {
+            ...baseAttrs,
+            status: ERROR,
+        })
+
+        throw new Error(`ContactExportTask with id = ${taskId} doesn't have value for "locale" field`)
+    }
+
+    try {
+        setLocaleForKeystoneContext(context, task.locale)
+
+        const allRoles = await ContactRole.getAll(context, {
+            deletedAt: null,
+            OR: [
+                { organization_is_null: true },
+                { organization: { id: where.organization.id } },
+            ],
+        })
+        const translatedRolesMap = Object.fromEntries(allRoles.map(role => ([role.id, role.name])))
+
+        const contactLoader = await buildContactsLoader({ where, sortBy })
+        const totalRecordsCount = await Contact.count(context, where)
+
+        const loadRecordsBatch = async (offset, limit) => {
+            const contacts = await contactLoader.loadChunk(offset, limit)
+
+            this.progress(Math.floor(offset / totalRecordsCount * 100))
+
+            return contacts
+        }
+
+        const convertRecordToFileRow = (contact) => contactToRow({
+            task, contact, translatedRolesMap,
+        })
+
+        switch (format) {
+            case EXCEL: {
```

Comment:
This is a bit confusing. Does this mean that there are other formats available?

Reply:
It will be added later, but not now